File: Example/VCSDK_for_media/ExampleVC/Root/SampleViewController.m
//
//  SampleViewController.m
//  ADLink
//
//  Created by Matsuki, Reiya (松木 玲也) on 2017/12/19.
//  Copyright © 2017年 ValueCommerce. All rights reserved.
//

#import "SampleViewController.h"

static NSString *kCellIdentifier = @"MyCellIdentifier";
static NSString *kTitleKey = @"title";
static NSString *kViewControllerClassNameKey = @"viewControllerClassName";

@interface SampleViewController()

@property (nonatomic, strong) NSArray *menuList;

@end

@implementation SampleViewController
- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.title = @"VCSDK_for_media Sample";
    
    self.menuList = @[@{kTitleKey: NSLocalizedString(@"LinkSwitch", @"LinkSwitch"),
                        kViewControllerClassNameKey: @"LSLinkSwitchViewController"},
                      @{kTitleKey: NSLocalizedString(@"UITextView_LinkSwitch", @"UITextView_LinkSwitch"),
                        kViewControllerClassNameKey: @"LSUITextViewController"},
                      @{kTitleKey: NSLocalizedString(@"UIWebView_LinkSwitch ", @"UIWebView_LinkSwitch "),
                        kViewControllerClassNameKey: @"LSUIWebViewController"},
                      @{kTitleKey: NSLocalizedString(@"WKWebView_LinkSwitch", @"WKWebView_LinkSwitch"),
                        kViewControllerClassNameKey: @"LSWKWebViewController"},
                      ];
    
    // create a custom navigation bar button and set it to always say "Back"
    UIBarButtonItem *temporaryBarButtonItem = [[UIBarButtonItem alloc] init];
    temporaryBarButtonItem.title = @"Back";
    self.navigationItem.backBarButtonItem = temporaryBarButtonItem;

}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    // this UIViewController is about to re-appear, make sure we remove the current selection in our table view
    NSIndexPath *tableSelection = [self.tableView indexPathForSelectedRow];
    [self.tableView deselectRowAtIndexPath:tableSelection animated:NO];
}

- (void)didReceiveMemoryWarning {
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc that aren't in use.
}


#pragma mark - Table view data source methods

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

// Customize the number of rows in the table view.
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return [self.menuList count];
}

// Customize the appearance of table view cells.
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kCellIdentifier];
    if (cell == nil)
    {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:kCellIdentifier];
        cell.accessoryType = UITableViewCellAccessoryDisclosureIndicator;
    }
    
    cell.textLabel.text = (self.menuList)[indexPath.row][kTitleKey];
    
    return cell;
}

#pragma mark - Table view delegate

- (UIViewController *)viewControllerWithDict:(NSDictionary*)dict
{
    NSString *controllerClassName = dict[kViewControllerClassNameKey];
    Class controllerClass = NSClassFromString(controllerClassName);
    UIViewController *viewController = nil;
    
    viewController = [[controllerClass alloc] initWithNibName:controllerClassName bundle:nil];
    
    if (!viewController.title) {
        viewController.title = dict[kTitleKey];
    }
    
    return viewController;
}

// Override to support row selection in the table view.
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSDictionary *targetViewControllerDict = (self.menuList)[indexPath.row];
    
    UIViewController *viewController = [self viewControllerWithDict:targetViewControllerDict];
    
    [[self navigationController] pushViewController:viewController animated:YES];
}

@end

File: Example/VCSDK_for_media_Example_Swift/ExampleVC/Root/SampleViewController.swift
//
//  SampleViewController.swift
//  ADLinkSwift
//
//  Created by Matsuki, Reiya (松木 玲也) on 2018/01/04.
//  Copyright © 2018年 ValueCommerce. All rights reserved.
//

import UIKit

class SampleViewController: UITableViewController {

    struct MenuItem {
        var name : String
        var className : String
    }
    
    let menuItems = [
        MenuItem(name: "LSLinkSwitch",               className: "LSLinkSwitchViewController"),
        MenuItem(name: "LinkSwitch_UItextView",      className: "LSUITextViewController"),
        MenuItem(name: "LinkSwitch_UIWebView",       className: "LSUIWebViewController"),
        MenuItem(name: "LinkSwitch_WKWebView",       className: "LSWKWebViewController"),
    ]
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - Table View
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return menuItems.count
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath) as UITableViewCell
        
        let menuItem = menuItems[indexPath.row]
        cell.textLabel?.text = menuItem.name
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        let menuItem = menuItems[indexPath.row]
        let viewControllerClass = NSClassFromString(NSString(format: "VCSDK_for_media_Example_Swift.%@", menuItem.className) as String) as? UIViewController.Type
        let viewController : UIViewController = viewControllerClass!.init(nibName: menuItem.className, bundle: nil)
        self.navigationController?.pushViewController(viewController, animated: true)
    }
}


